lib/index.js

```diff
@@ -19,6 +90,57 @@ function handleTokenRequest({logger}) {
     }
   };
 }
-odule.exports = {
-  handleTokenRequest
+
+function _authenticateClient({client, clientSecret}) {
+  if(!client) {
+    throw new InvalidClient({
+      description: 'Unknown client identifier.'
+    });
+  }
+
+  const storedClientSecret = client.clientSecret || client.client_secret;
+
+  if(clientSecret !== storedClientSecret) {
+    throw new InvalidClient({
+      description: 'Mismatched client secret.'
+    });
+  }
+}
+
+function _validateParams({clientId, clientSecret, grantType, scope}) {
+  if(!clientId || !clientSecret) {
+    throw new UnauthorizedClient({
+      description: 'Missing client credentials at the Token endpoint.'
+    });
+  }
+  if(!grantType) {
+    throw new InvalidRequest({
+      description: 'Missing grant_type parameter.'
+    });
+  }
+  if(!scope) {
+    throw new InvalidRequest({
+      description: 'Missing scope parameter.'
+    });
+  }
+  if(grantType !== 'client_credentials') {
+    // Fixme: should be unsupported_grant_type error
```

Comment:
can this be addressed?

Reply:
Let's open an issue for it, it's more of a wishlist item (it's a more specific error, but the current one is fine).

Reply:
n/m, it's addressed.